packages/plugins/src/plugins/nerit/getVaultsPosition.ts

```diff
@@ -0,0 +1,52 @@
+import { EvmNetworkIdType } from '@sonarwatch/portfolio-core';
+import { getAddress } from 'viem';
+import { getEvmClient } from '../../utils/clients';
+import { EnrichedTroveInfos } from './types';
+
+import { trovesAbi } from './abi';
+
+export async function getVaultsPositions(
+  networkId: EvmNetworkIdType,
+  vaults: EnrichedTroveInfos[]
+) {
+  const client = getEvmClient(networkId);
+  const flatVaults = vaults.flatMap((vault) =>
+    vault.collateral.troves.map((trove) => ({
+      manager: getAddress(vault.manager),
+      collateral: vault.collateral,
+      collateralTokenAddress: vault.collateralTokenAddress,
+      id: trove.id,
+    }))
+  );
+
+  const contracts = flatVaults.map(
+    (trove) =>
+      ({
+        address: trove.manager,
+        abi: trovesAbi,
+        functionName: 'Troves',
+        // id of the NFT corresponding to the trove (format : "2:0xc75c80fc611e5ebfb43d1341c14bcabee1dd371a2776074180e0916dcf7d8813")
+        args: [trove.id.split(':')[1]],
+      } as const)
```

Comment:
i think you can do this so you dont need to cast below and get better typesafety on args

```suggestion
  const contracts = flatVaults.map(
    (trove) =>
      ({
        address: trove.manager,
        abi: trovesAbi,
        functionName: 'Troves',
        // id of the NFT corresponding to the trove (format : "2:0xc75c80fc611e5ebfb43d1341c14bcabee1dd371a2776074180e0916dcf7d8813")
        args: [trove.id.split(':')[1]],
      } satisfies ContractFunctionConfig<typeof trovesAbi>)
```